doc/source/how-to-use-middleware.rst

```diff
@@ -0,0 +1,85 @@
+Use Built-in Middleware Layers
+==============================
+
+In this tutorial, we will learn how to utilize built-in middleware layers to augment the behavior of a flower callable. Middleware allows us to perform operations before and after a task is processed in the flower callable.
+
+What is Middleware?
+-------------------
+
+Middleware is a callable that wraps around a Flower Callable. It can manipulate or inspect incoming tasks (``TaskIns``) in the ``Fwd`` and the resulting tasks (``TaskRes``) in the ``Bwd``. The signature for a middleware layer (``Layer``) is as follows:
+
+.. code-block:: python
+
+    FlowerCallable = Callable[[Fwd], Bwd]
+    Layer = Callable[[Fwd, FlowerCallable], Bwd]
+
+A typical middleware function might look something like this:
+
+.. code-block:: python
+
+    def example_middleware(fwd: Fwd, fc: FlowerCallable) -> Bwd:
+        # Do something with Fwd before passing to the inner flower callable.
+        bwd = fc(fwd)
+        # Do something with Bwd before returning.
+        return bwd
+
+Using Middleware Layers
+-----------------------
+
+To use middleware layers in your flower callable, you can follow these steps:
+
+1. Import the Required Middleware
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+First, import the built-in middleware layers you intend to use:
+
+.. code-block:: python
+
+    import flwr as fl
+    from flwr.client.middleware import example_middleware1, example_middleware2
```

Comment:
Can we do sth different here? This example won't work if anyone were to try it

Reply:
We don't have any built-in middleware for now. One alternative is to write simple `example_middleware1` and `example_middleware2` functions in the doc and use them. Wdyt?